### DISCUSSION TITLE
BCUT2D descriptors are NaN

### DISCUSSION THREAD

**Discussion by lisa0895** [rdkit/rdkit#5824] [Q&A]
I have the molecule sodium mercaptobenzothiazole (SMILES: C1=CC=C2C(=C1)N=C(S2)[S-].[Na+]) and want to generate molecular descriptors for it using: 

```
smiles = 'C1=CC=C2C(=C1)N=C(S2)[S-].[Na+]'
mol = Chem.MolFromSmiles(smiles)
calc = MoleculeDescriptors.MolecularDescriptorCalculator([x[0] for x in Descriptors._descList])
header = calc.GetDescriptorNames()
ds = [calc.CalcDescriptors(mol)]
df = pd.DataFrame(ds, columns=header)
```

The Problem here is, that I receive NaN for the BCUT2D descriptors. What could be the reason that it does not work for sodium mercaptobenzothiazole?
Thank you in advance!

Edit: This problem seems to occur for all SMILES containing squared brackets, for example [Li] or [Na+].[Cl-]

**Comment by bp-kelley** [comment]
Thanks for the example, it looks like sodium makes it explode:

```
Descriptors.BCUT2D_CHGHI(Chem.MolFromSmiles("[Na+]"))
nan
```

I'll do some investigation on this one, we've seen similar things before with the CHGHI and CHGLO

**Reply by bp-kelley** [reply]
Others seem to be okay:

```
>>> Descriptors.BCUT2D_CHGHI(Chem.MolFromSmiles("[S-]"))
-1.0
>>> Descriptors.BCUT2D_CHGHI(Chem.MolFromSmiles("[Cl-]"))
-1.0
```

**Reply by greglandrum** [reply]
@bp-kelley tracked this down and the reason this is happening is that there aren't Gasteiger charges for `[Na+]`.

You'll get NaN for any molecule which contains an atom that isn't parameterized for the Gasteiger charges.

The way to handle this is to detect NaN in your code and either fail for the molecule or add a default value that works for your application. You can best decide what the right answer there is based on what you are planning on doing with the descriptors.

**Comment by thegodone** [comment]
I have same issue with Ge, Sn etc… metals.Envoyé de mon iPhoneLe 19 janv. 2023 à 16:25, Brian Kelley ***@***.***> a écrit :﻿
Others seem to be okay:
>>> Descriptors.BCUT2D_CHGHI(Chem.MolFromSmiles("[S-]"))
-1.0
>>> Descriptors.BCUT2D_CHGHI(Chem.MolFromSmiles("[Cl-]"))
-1.0

**Comment by canozkan42** [comment]
Removing salt anions before calculating molecular descriptors also helps, organic salts always seem to have such calculation issues
